tests/expectations/annotation.cpp

```diff
@@ -53,16 +54,16 @@ union F {
 
   static F Foo(const int16_t &a0) {
     F result;
-    result.foo._0 = a0;
+    new (&result.foo._0) int16_t(a0);
     result.tag = Tag::Foo;
     return result;
   }
 
   static F Bar(const uint8_t &aX,
                const int16_t &aY) {
     F result;
-    result.bar.x = aX;
-    result.bar.y = aY;
+    new (&result.bar.x) uint8_t(aX);
+    new (&result.bar.y) int16_t(aY);
```

Comment:
As more of a general note, I'm pretty sure you could technically placement new the entire structure at once using direct aggregate initialization. Keeping it this way could be cleaner though, due to the trickiness involved with fixed sized arrays.

```c++
new (&result.bar) Bar_Body{aX, aY};
```

Reply:
Yeah, I'll keep it like this for now.